framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
Private ones are not meant to be set from the input file so they shouldn't show up in Peacock. We could do something different for the developer API though. I'm all for documenting these things better. Let's continue that discussion on a different issue or the documentation issue @aeslaughter is working on.